connector/src/main/scala/com/microsoft/kusto/spark/datasource/KustoRelation.scala

```diff
@@ -45,7 +50,15 @@ private[kusto] case class KustoRelation(kustoCoordinates: KustoCoordinates,
 
   override def buildScan(): RDD[Row] = {
     val kustoClient = KustoClientCache.getClient(kustoCoordinates.cluster, authentication).engineClient
-    val count = KDSU.countRows(kustoClient, query, kustoCoordinates.database)
+
+    var count = 0
+    try {
+      count = Await.result(Future(KDSU.countRows(kustoClient, query, kustoCoordinates.database)), timeout)
```

Comment:
where is `timeout` set?

Reply:
timeout is set by user - but i replaced the usage here with timeoutForCountCheck ..